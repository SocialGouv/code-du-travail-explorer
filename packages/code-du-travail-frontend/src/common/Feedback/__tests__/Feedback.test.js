import React from "react";
import { render } from "react-testing-library";

import { Feedback } from "..";

jest.mock("react-piwik", () => {
  let events = [];
  return {
    events,
    push: event => events.push(event),
    flush: () => {
      events = [];
    }
  };
});
const ReactPiwik = require("react-piwik");

afterEach(ReactPiwik.flush);

describe("<Feedback/>", () => {
  it("should render", () => {
    const { container } = render(
      <Feedback
        query="Query"
        source="Fiches"
        url="http://server/fiches/fiche1"
      />
    );
    expect(container).toMatchSnapshot();
  });
  it("should render form once user answer yes", () => {
    const { container, getByText } = render(
      <Feedback
        query="Query"
        source="Fiches"
        url="http://server/fiches/fiche1"
      />
    );
    const button = getByText(/oui/i);
    button.click();
    expect(container).toMatchSnapshot();
  });
  it("should render form once user answer no", () => {
    const { container, getByText } = render(
      <Feedback
        query="Query"
        source="Fiches"
        url="http://server/fiches/fiche1"
      />
    );
    const button = getByText(/non/i);
    button.click();
    expect(container).toMatchSnapshot();
  });
  it("should send piwik event when click yes", () => {
    const { getByText } = render(
      <Feedback
        query="Query"
        source="Fiches"
        url="http://server/fiches/fiche1"
      />
    );
    const button = getByText(/oui/i);
    button.click();
    expect(JSON.stringify(ReactPiwik.events)).toMatchSnapshot();
    ReactPiwik.flush();
  });
  it("should send piwik event when click no", () => {
    const { getByText } = render(
      <Feedback
        query="Query"
        source="Fiches"
        url="http://server/fiches/fiche1"
      />
    );
    const button = getByText(/non/i);
    button.click();
    expect(JSON.stringify(ReactPiwik.events)).toMatchSnapshot();
  });
});
