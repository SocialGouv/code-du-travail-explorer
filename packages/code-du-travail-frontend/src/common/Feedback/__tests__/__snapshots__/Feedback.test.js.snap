// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Feedback/> should render 1`] = `
.c1 {
  max-width: 1200px;
  margin: 0 auto;
  padding: 0 1.25rem;
}

.c1 > *:first-child {
  margin-top: 0;
}

.c1 > *:last-child {
  margin-bottom: 0;
}

.c0 {
  padding: 1.25rem 0;
  color: #434956;
}

.c2 {
  padding: 0.625rem 1.25rem;
  color: #434956;
  border: 1px solid #c9d3df;
  border-radius: 0.25rem;
  background-color: #ebeff3;
}

.c2 > *:first-child {
  margin-top: 0;
}

.c2 > *:last-child {
  margin-bottom: 0;
}

.c5 {
  padding: 0.625rem 1.5rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  text-align: center;
  line-height: inherit;
  font-size: 1rem;
  font-weight: 600;
  vertical-align: middle;
  border: 1px solid;
  border-radius: 1.5rem;
  cursor: pointer;
  -webkit-transition: background-color 250ms ease;
  transition: background-color 250ms ease;
  color: #0c0c0e;
  background: #ebeff3;
  border-color: #ebeff3;
}

.c5:not([disabled]):hover,
.c5:not([disabled]):active,
.c5:not([disabled]):focus {
  background: #fbfcfd;
  color: #18181c;
}

.c5[disabled] {
  cursor: not-allowed;
  color: rgba(12,12,14,0.4);
}

.c8 {
  padding: 0.625rem 1.5rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  text-align: center;
  line-height: inherit;
  font-size: 1rem;
  font-weight: 600;
  vertical-align: middle;
  border: 1px solid;
  border-radius: 1.5rem;
  cursor: pointer;
  -webkit-transition: background-color 250ms ease;
  transition: background-color 250ms ease;
  color: #fff;
  background: #006be6;
  border-color: #006be6;
}

.c8:not([disabled]):hover,
.c8:not([disabled]):active,
.c8:not([disabled]):focus {
  background: #07f;
  color: #fff;
}

.c8[disabled] {
  cursor: not-allowed;
  color: rgba(255,255,255,0.4);
}

.c3 {
  margin-bottom: 1rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

.c7 {
  min-width: 7rem;
}

.c7 + .c6 {
  margin-left: 1rem;
}

@media print {
  .c1 {
    max-width: 100%;
    padding: 0;
  }
}

@media (max-width:600px) {
  .c2 {
    padding: 0.625rem 1.25rem;
  }
}

@media print {
  .c2 {
    border: none;
    padding: 0 5pt;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h3
          class="c3"
        >
          Avez-vous trouvé la réponse à votre question ?
        </h3>
        <p
          class="c4"
        >
          <button
            class="c5 c6 c7"
          >
            Non
          </button>
          <button
            class="c8 c6 c7"
          >
            Oui
          </button>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Feedback/> should render form once user answer no 1`] = `
.c1 {
  max-width: 1200px;
  margin: 0 auto;
  padding: 0 1.25rem;
}

.c1 > *:first-child {
  margin-top: 0;
}

.c1 > *:last-child {
  margin-bottom: 0;
}

.c0 {
  padding: 1.25rem 0;
  color: #434956;
}

.c2 {
  padding: 0.625rem 1.25rem;
  color: #434956;
  border: 1px solid #c9d3df;
  border-radius: 0.25rem;
  background-color: #ebeff3;
}

.c2 > *:first-child {
  margin-top: 0;
}

.c2 > *:last-child {
  margin-bottom: 0;
}

.c7 {
  padding: 0.625rem 1.5rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  text-align: center;
  line-height: inherit;
  font-size: 1rem;
  font-weight: 600;
  vertical-align: middle;
  border: 1px solid;
  border-radius: 1.5rem;
  cursor: pointer;
  -webkit-transition: background-color 250ms ease;
  transition: background-color 250ms ease;
  color: #fff;
  background: #005994;
  border-color: #005994;
}

.c7:not([disabled]):hover,
.c7:not([disabled]):active,
.c7:not([disabled]):focus {
  background: #0068ae;
  color: #fff;
}

.c7[disabled] {
  cursor: not-allowed;
  color: rgba(255,255,255,0.4);
}

.c9 {
  padding: 0.625rem 1.5rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  text-align: center;
  line-height: inherit;
  font-size: 1rem;
  font-weight: 600;
  vertical-align: middle;
  border: 1px solid;
  border-radius: 1.5rem;
  cursor: pointer;
  -webkit-transition: background-color 250ms ease;
  transition: background-color 250ms ease;
  padding: 0;
  vertical-align: initial;
  text-align: left;
  line-height: initial;
  color: #005994;
  font-weight: normal;
  font-size: inherit;
  background: none;
  border: none;
  border-radius: 0;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c9:focus,
.c9:hover,
.c9:active {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.c5 {
  margin-bottom: 1.25rem;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c8 {
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  margin: 0;
  margin-top: 1.25rem;
}

.c3 {
  margin-bottom: 1rem;
}

@media print {
  .c1 {
    max-width: 100%;
    padding: 0;
  }
}

@media (max-width:600px) {
  .c2 {
    padding: 0.625rem 1.25rem;
  }
}

@media print {
  .c2 {
    border: none;
    padding: 0 5pt;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h3
          class="c3"
        >
          Avez-vous trouvé la réponse à votre question ?
        </h3>
        <form
          action="https://formspree.io/xbalrgzm"
          class="c4"
        >
          <p>
            <strong>
              Merci pour votre réponse !
            </strong>
          </p>
          <p>
            Vous souhaitez nous aider à améliorer le Code du travail numérique ? Laissez-nous un commentaire :
          </p>
          <input
            name="question"
            type="hidden"
            value="Query"
          />
          <input
            name="url"
            type="hidden"
            value="http://server/fiches/fiche1"
          />
          <input
            name="sourceType"
            type="hidden"
            value="Fiches Service Public"
          />
          <select
            class="c5"
            name="motif"
          >
            <option
              value="Les informations proposées ne sont pas à jour"
            >
              Les informations proposées ne sont pas à jour
            </option>
            <option
              value="Les informations me paraissent erronées"
            >
              Les informations me paraissent erronées
            </option>
            <option
              value="Les informations me paraissent incomplètes"
            >
              Les informations me paraissent incomplètes
            </option>
            <option
              value="Les informations me paraissent hors-sujet"
            >
              Les informations me paraissent hors-sujet
            </option>
            <option
              value="Je n'ai pas trouvé la réponse à ma question"
            >
              Je n'ai pas trouvé la réponse à ma question
            </option>
          </select>
          <textarea
            class="c5"
            name="message"
            placeholder="Les informations ..."
          />
          <div
            class="c6"
          >
            <button
              class="c7"
            >
              Envoyer mon commentaire
            </button>
            <div
              class="c8"
            />
          </div>
        </form>
        <p>
          Pour obtenir une réponse à votre question de droit du travail, nous vous invitons à joindre les
           
          <button
            class="c9"
          >
            services de renseignement
          </button>
          .
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Feedback/> should render form once user answer yes 1`] = `
.c1 {
  max-width: 1200px;
  margin: 0 auto;
  padding: 0 1.25rem;
}

.c1 > *:first-child {
  margin-top: 0;
}

.c1 > *:last-child {
  margin-bottom: 0;
}

.c0 {
  padding: 1.25rem 0;
  color: #434956;
}

.c2 {
  padding: 0.625rem 1.25rem;
  color: #434956;
  border: 1px solid #c9d3df;
  border-radius: 0.25rem;
  background-color: #ebeff3;
}

.c2 > *:first-child {
  margin-top: 0;
}

.c2 > *:last-child {
  margin-bottom: 0;
}

.c7 {
  padding: 0.625rem 1.5rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  text-align: center;
  line-height: inherit;
  font-size: 1rem;
  font-weight: 600;
  vertical-align: middle;
  border: 1px solid;
  border-radius: 1.5rem;
  cursor: pointer;
  -webkit-transition: background-color 250ms ease;
  transition: background-color 250ms ease;
  color: #fff;
  background: #005994;
  border-color: #005994;
}

.c7:not([disabled]):hover,
.c7:not([disabled]):active,
.c7:not([disabled]):focus {
  background: #0068ae;
  color: #fff;
}

.c7[disabled] {
  cursor: not-allowed;
  color: rgba(255,255,255,0.4);
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.c5 {
  margin-bottom: 1.25rem;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c8 {
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  margin: 0;
  margin-top: 1.25rem;
}

.c3 {
  margin-bottom: 1rem;
}

@media print {
  .c1 {
    max-width: 100%;
    padding: 0;
  }
}

@media (max-width:600px) {
  .c2 {
    padding: 0.625rem 1.25rem;
  }
}

@media print {
  .c2 {
    border: none;
    padding: 0 5pt;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h3
          class="c3"
        >
          Avez-vous trouvé la réponse à votre question ?
        </h3>
        <form
          action="https://formspree.io/xbalrgzm"
          class="c4"
        >
          <p>
            <strong>
              Merci pour votre réponse !
            </strong>
          </p>
          <p>
            Vous souhaitez nous aider à améliorer le Code du travail numérique ? Laissez-nous un commentaire :
          </p>
          <input
            name="question"
            type="hidden"
            value="Query"
          />
          <input
            name="url"
            type="hidden"
            value="http://server/fiches/fiche1"
          />
          <input
            name="sourceType"
            type="hidden"
            value="Fiches Service Public"
          />
          <textarea
            class="c5"
            name="message"
            placeholder="Les informations ..."
          />
          <div
            class="c6"
          >
            <button
              class="c7"
            >
              Envoyer mon commentaire
            </button>
            <div
              class="c8"
            />
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Feedback/> should send piwik event when click no 1`] = `"[]"`;

exports[`<Feedback/> should send piwik event when click yes 1`] = `"[]"`;
